src/app/ReadClient.cpp

```diff
@@ -717,5 +794,37 @@ CHIP_ERROR ReadClient::SendSubscribeRequest(ReadPrepareParams & aReadPreparePara
     return CHIP_NO_ERROR;
 }
 
-}; // namespace app
-}; // namespace chip
+void ReadClient::OnResubscribeTimerCallback(System::Layer * apSystemLayer, void * apAppState)
+{
+    ReadClient * const _this = reinterpret_cast<ReadClient *>(apAppState);
+    assert(_this != nullptr);
+    _this->SendSubscribeRequest(_this->mReadPrepareParams);
+    _this->mNumRetries++;
+}
+
+bool ReadClient::Resubscribe()
+{
+    bool shouldResubscribe = true;
+    mReadPrepareParams.mResubscribePolicy(mNumRetries, mIntervalMsec, shouldResubscribe);
```

Comment:
Would it make sense for the null-check for this function pointer to just live inside this function, along with the other checks we do here?

Reply:
agree, done, thanks